firebase/functions/index.js

```diff
@@ -5,12 +5,23 @@ const functions = require('firebase-functions')
 const admin = require('firebase-admin')
 admin.initializeApp()
 
-// Increments or decrements various counters of User and Group once new reults are pushed.
+
+// Calculate number of users who finished a group and
```

Comment:
Can you add an overview explanation of how the `requiredCount` and all get updated via `groupUsers`? I think it's not immediately obvious when reading the code below.